libraries/AP_Scripting/modules/mavport.lua

```diff
@@ -0,0 +1,182 @@
+                new_buf[#new_buf+1] = s
+                new_len = new_len + #s
+            else
+                for ci = 1, #s do
+                    if count == SERIAL_CONTROL_DATA_LEN then -- re-buffer
+                        s = s:sub(ci)
+                        new_buf[#new_buf+1] = s
+                        new_len = new_len + #s
+                        break
+                    end
+                    count = count + 1
+                    data[count] = s:byte(ci)
+                end
+            end
+        end
+        for ci = count+1, SERIAL_CONTROL_DATA_LEN do -- zero unused bytes
+            data[ci] = 0
+        end
+        self._tx_template.count = count
+
+        -- store remaining buffered messages
+        self._tx_buf = new_buf
+        self._tx_buf_len = new_len
+
+        -- encode the message and store it for transmission
+        _, self._tx_msg = mavlink_msgs.encode(
+            "SERIAL_CONTROL", self._tx_template)
+    end
+
+    if self._tx_msg then -- message to send
+        if mavlink.send_chan(self._chan, SERIAL_CONTROL.id, self._tx_msg) then
+            self._tx_msg = nil -- successfully sent
+        end
+    end
+end
+
+function mavport:read()
+    if #self._rx_buf == 0 then
+        self:_receive()
+        if #self._rx_buf == 0 then return -1 end
+    end
+
+    local s = self._rx_buf[1]
+    local pos = self._rx_buf_pos
+    local c = s:byte(pos)
+
+    if pos == #s then
+        table.remove(self._rx_buf, 1)
+        pos = 0
+    end
+    self._rx_buf_pos = pos + 1
+
+    return c 
+end
+
+function mavport:readstring(count)
+    -- TODO: implement more efficiently if we actually need it
+    local ret = {}
+    while count > 0 do
+        local c = self:read()
+        if c == -1 then break end
+        ret[#ret+1] = c
+        count = count - 1
+    end
+
+    return string.byte(table.unpack(ret))
+end
+
+function mavport:_receive()
+    local msg, chan
+    while true do
+        msg, chan = mavlink.receive_chan()
+        if not msg then return end -- no new messages
+
+        -- decode message and handle if it's for us
+        msg = mavlink_msgs.decode(msg, msg_map)
+        if msg.device == 10 then -- for the dev shell?
+            self._chan = chan -- reply on the same channel
+            break
+        end
+    end
+
+    -- remove trailing nulls, they shouldn't happen but they do
+    while msg.data[msg.count] == 0 do
+        msg.count = msg.count - 1
+    end
+
+    -- store the data into the rx buffer as a string
+    local s = string.char(table.unpack(msg.data, 1, msg.count))
+    if #s > 0 then
+        self._rx_buf[#self._rx_buf+1] = s
+    end
+end
+
+function mavport:available()
+    if #self._rx_buf == 0 then
+        self:_receive()
+    end
+
+    local avail = 0
+    for _, b in ipairs(self._rx_buf) do
+        avail = avail + #b
+    end
+
+    return avail
+end
+
+-- for completeness
+function mavport:set_flow_control(flow_control_setting) -- luacheck: ignore 212
```

Comment:
```suggestion
function mavport:set_flow_control(_)
```
If you use a underscore you can tell it that your ignoring the argument. It might be enough to just prefix with a underscore. I don't recall.

Reply:
Apparently on the version I have an underscore prefix is enough, but not so with the one in CI.